datalad/interface/common_opts.py

```diff
@@ -148,4 +148,14 @@
     [PY: Multiple dependencies can be given as a list of sibling names
     PY][CMD: This option can be given more than once to configure multiple
     dependencies CMD]""",
+    action='append',
+    constraints=EnsureStr() | EnsureNone())
+
+publish_by_default = Parameter(
+    args=("--publish-by-default",),
+    metavar='REFSPEC',
+    doc="""add a refspec to be published to this sibling by default if nothing
+    specified.""",
+    constraints=EnsureStr() | EnsureNone(),
     action='append')
```

Comment:
as discussed briefly during hangout -- information about what to be pushed shouldn't go to `.datalad/config` since that one contains persistent/committed to git information and we are just talking about specific siblings (remotes) of a specific dataset (clone), so imho should be just in `.git/config`,.

But there specifically -- how do I to specify to push both all refs and all tags?? should then at least be `nargs="+"` and me to specify lengthy `--publish-by-default=refs/tags/* --publish-by-default=refs/heads/*`, or how?